app/src/main/java/com/simformsolutions/sample/app/ui/main/Repositories.kt

```diff
@@ -31,155 +31,161 @@ import androidx.compose.foundation.layout.fillMaxWidth
 import com.simformsolutions.sample.app.utils.GITHUB_UPDATED_AT_TIMESTAMP
+import com.simformsolutions.sample.app.utils.extension.showToast
+import com.simformsolutions.sample.app.utils.extension.toColor
 import kotlinx.coroutines.flow.Flow
 import java.text.SimpleDateFormat
 import java.util.*
 
-@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun Repositories(
     repositoryData: Flow<PagingData<RepositoriesQuery.Node>>
 ) {
+    var shouldShowInitialLoader by remember { mutableStateOf(false) }
     val repositories = repositoryData.collectAsLazyPagingItems()
-    val shouldShowCircularProgressBar by remember {
-        derivedStateOf { repositories.itemCount == 0 }
+    when (val currentState = repositories.loadState.refresh) {
+        is LoadState.Error ->
+            LocalContext.current.showToast(
+                stringResource(
+                    R.string.cannot_load_repositories,
+                    currentState.error.message.toString()
+                ),
+                Toast.LENGTH_LONG
+            )
+        else -> shouldShowInitialLoader = currentState is LoadState.Loading
     }
 
-    Scaffold(
-        topBar = { TopAppBar() }
-    ) { padding ->
-        Box(
+    Box {
+        CircularProgressIndicator(
             modifier = Modifier
-                .padding(padding)
-                .fillMaxSize()
-        ) {
-            CircularProgressIndicator(
-                modifier = Modifier.align(Alignment.Center),
-                isHidden = !shouldShowCircularProgressBar
-            )
-            RepositoriesList(
-                modifier = Modifier.fillMaxSize(),
-                repositories = repositories
-            )
-        }
+                .align(Alignment.Center)
+                .alpha(if (shouldShowInitialLoader) 1f else 0f)
+        )
+        RepositoriesList(
+            modifier = Modifier.fillMaxSize(),
+            repositories = repositories
+        )
     }
 }
 
-@OptIn(ExperimentalMaterial3Api::class)
-@Composable
-private fun TopAppBar() {
-    TopAppBar(
-        title = {
-            Text(
-                text = stringResource(R.string.simform_repo_title),
-                fontSize = 20.sp,
-                fontWeight = FontWeight.Bold,
-                color = topAppBarTitleColor
-            )
-        },
-        modifier = Modifier.shadow(7.dp)
-    )
-}
-
-@Composable
-private fun CircularProgressIndicator(
-    modifier: Modifier = Modifier,
-    isHidden: Boolean = true
-) {
-    CircularProgressIndicator(
-        modifier = modifier
-            .alpha(if (isHidden) 0f else 1f)
-    )
-}
-
 @Composable
 private fun RepositoriesList(
     modifier: Modifier = Modifier,
     repositories: LazyPagingItems<RepositoriesQuery.Node>
 ) {
-    val uriHandler = LocalUriHandler.current
+    var shouldShowRepositoryPageLoader by remember { mutableStateOf(false) }
+    when (val currentState = repositories.loadState.append) {
```

Comment:
This logic is repetitive in `Repositories`. Lets generalise it so that we can reuse it.